### PR TITLE
feat(react-input): Add optional shadow to filled appearances

### LINKED ISSUE #24642: Inputs: Add optional shadows to each component
- [ ] Textarea
  - issue: #24512
- [ ] Input
  - issue: #24730
- [ ] Spinbutton
  - issue:
- [ ] Select
  - issue:
- [ ] Combobox
  - issue:

### PR DESCRIPTION
**Pull request by sopranopillow**
image

## Related Issue(s)

Fixes #24642

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/a07747ceae4ac6c360674ee0a2d6cdf61efe6164'>a07747ceae4ac6c360674ee0a2d6cdf61efe6164</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24730/perf-test-react-components/Avatar-mount_base.html">1737</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24730/perf-test-react-components/Avatar-mount.html">1755</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24730/perf-test-react-components/Button-mount_base.html">1228</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24730/perf-test-react-components/Button-mount.html">1200</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24730/perf-test-react-components/FluentProvider-mount_base.html">2023</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24730/perf-test-react-components/FluentProvider-mount.html">1985</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24730/perf-test-react-components/FluentProviderWithTheme-mount_base.html">759</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24730/perf-test-react-components/FluentProviderWithTheme-mount.html">768</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24730/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">690</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24730/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">727</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24730/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">766</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24730/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">768</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24730/perf-test-react-components/MakeStyles-mount_base.html">2392</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24730/perf-test-react-components/MakeStyles-mount.html">2366</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24730/perf-test-react-components/SpinButton-mount_base.html">3377</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24730/perf-test-react-components/SpinButton-mount.html">3329</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-input/etc/react-input.api.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ export type InputOnChangeData = {
 export type InputProps = Omit<ComponentProps<Partial<InputSlots>, 'input'>, 'children' | 'defaultValue' | 'onChange' | 'size' | 'type' | 'value'> & {
     children?: never;
     size?: 'small' | 'medium' | 'large';
-    appearance?: 'outline' | 'underline' | 'filled-darker' | 'filled-lighter';
+    appearance?: 'outline' | 'underline' | 'filled-darker' | 'filled-lighter' | 'filled-darker-shadow' | 'filled-lighter-shadow';
```
Comment (micahgodbolt) [inline packages/react-components/react-input/etc/react-input.api.md:29]: are these new names standardized across controls? feels like our appearance list is a drop zone for any new visual variant design asks for.

Comment (micahgodbolt) [reply packages/react-components/react-input/etc/react-input.api.md:29]: okay, i see this is work across multiple controls

https://github.com/microsoft/fluentui/issues/24642

Comment (sopranopillow) [reply packages/react-components/react-input/etc/react-input.api.md:29]: I'm in the process of adding it to all input components which are the only components that have filled-lighter/filled-darker. So all of them will have the same names.

**Review by micahgodbolt — APPROVED** [review]

**Review by khmakoto — APPROVED** [review]

**Pull request merged**